client/state/data-layer/wpcom/read/tags/index.js

```diff
@@ -32,42 +35,30 @@ export function requestTags( store, action, next ) {
 	next( action );
 }
 
-/**
- * Normalize response from the api so whether we get back a single tag or a list of tags
- * we always pass forward a list
- * @param  {Tag|Tags} apiResponse api response from the tags endpoint
- * @return {Tags}             An object containing list of tags
- */
-function fromApi( apiResponse ) {
-	let tags;
-	if ( apiResponse.tag )	 {
-		tags = [ apiResponse.tag ];
-	} else if ( apiResponse.tags ) {
-		tags = map( apiResponse.tags, tag => ( { ...tag, is_following: true } ) );
-	} else {
-		if ( process.env.NODE_ENV === 'development' ) {
-			throw new Error( 'bad api response for /read/tags' );
-		}
-		tags = [];
+export function receiveTagsSuccess( store, action, next, apiResponse ) {
+	let tags = fromApi( apiResponse );
+	// if from the read/tags api, then we should add isFollowing=true to all of the tags
+	if ( apiResponse.tags ) {
```

Comment:
^ I think it _is_ the evidence.  Theres not much more (not that this is insignificant).

It'll be pretty easy to separate out later, but I think the way things currently end up between the reducer and the data-layer isn't bad